app/models/filter.rb

```diff
@@ -28,6 +28,7 @@ def self.add_current_location?
   belongs_to :role
   has_many :filterings, :dependent => :destroy
   has_many :permissions, :through => :filterings
+  has_many :filters
```

Comment:
If we keep filters linking other filters, please give it some more descriptive name, e.g. `limiting_filters`